spec/models/metric/ci_mixin/capture_spec.rb

```diff
@@ -36,6 +36,35 @@ def expected_stats_period_end
     parse_datetime('2013-08-28T12:41:40Z')
   end
 
+  context "#perf_capture_queue" do
+    it "splits up long perf_capture durations for old last_perf_capture_on" do
+      # there should be 10 days + a partial day, or 11 queue items
+      Timecop.freeze
+      original_queue_count = MiqQueue.count
+      start_time = 10.days.ago - 5.hours - 23.minutes
+      @vm.last_perf_capture_on = start_time
+      @vm.perf_capture_queue("realtime")
+      expect(MiqQueue.count - original_queue_count).to eq 11
+
+      # make sure the queue items are in the correct order
+      queue_items = MiqQueue.peek(:conditions => {:zone       => @vm.my_zone,
+                                                  :queue_name => @vm.queue_name_for_metrics_collection,
+                                                  :role       => "ems_metrics_collector"},
+                                  :limit      => 3)
+      start_time = (start_time + 10.days).utc
+      end_time = (start_time + 5.hours + 23.minutes).utc
+      queue_items.each do |q_item|
+        q_start_time, q_end_time = q_item.args
+        expect(q_start_time).to eq start_time
+        expect(q_end_time).to eq end_time
```

Comment:
You *might* want to use `be_same_time_as` for the times here for start and end time.  It's possible that since we are serializing that there could be a loss of precision which would given sporadic test failures.